### DISCUSSION TITLE
Watch changes from an endpoint

### DISCUSSION THREAD

**Discussion by kitze** [dotansimha/graphql-code-generator#5580] [Q&A] [answered]
I was wondering if there's a way to setup `graphql-code-generator` so it can watch an endpoint (i.e localhost:4000) for changes, so when there are changes in the schema, the TS types and the `schema.graphql` file would be automatically generated.

**Comment by dotansimha** [comment, marked as answer]
Hi @kitze ! :) 
Currently, we don't have a way to watch endpoints. We do have a flexible `watch` configuration that allows you to add a list of custom files you need to watch, but it's currently doesn't support endpoints.
I think the only way to do that is to run GraphQL introspection over the endpoint, using polling, and then check if the schema has changed? I'm not sure how it will behave with large schemas, but we can definitely give it a try.
I created an issue for tracking: https://github.com/dotansimha/graphql-code-generator/issues/5609
